File: web/src/lib/stores/tts.ts
import { writable } from 'svelte/store';

export type TTSEvent = {
	voice: string;
	voice_label?: string;
	text: string;
	requested_by: string;
	platform: string;
	channel_id: string;
	timestamp: string;
	audio_base64?: string;
};

const createQueue = () => {
	const { subscribe, update } = writable<TTSEvent[]>([]);

	return {
		subscribe,
		push: (event: TTSEvent) => {
			update((list) => [event, ...list].slice(0, 50));
		}
	};
};

export const ttsQueue = createQueue();

File: internal/domain/tts.go
package domain

import (
	"context"
	"time"
)

type TTSEvent struct {
	Voice       string    `json:"voice"`
	VoiceLabel  string    `json:"voice_label,omitempty"`
	Text        string    `json:"text"`
	RequestedBy string    `json:"requested_by"`
	Platform    Platform  `json:"platform"`
	ChannelID   string    `json:"channel_id"`
	Timestamp   time.Time `json:"timestamp"`
	AudioBase64 string    `json:"audio_base64"`
}

type TTSEventPublisher interface {
	PublishTTSEvent(ctx context.Context, event TTSEvent) error
}

type TTSSettingsRepository interface {
	SetTTSVoice(ctx context.Context, voice string) error
	GetTTSVoice(ctx context.Context) (string, error)
}
